### PR TITLE
Service reliability sli slo

### PR DESCRIPTION
**Pull request by ivorc**

### CODE REVIEW

File: practices/service-reliability.md [hunk 1]
```diff
@@ -8,8 +8,12 @@
...
-    * Agree an incident severity classification and the response which each level requires in terms of communication and resolution (example: [HSCN Consumer Handbook](https://digital.nhs.uk/services/health-and-social-care-network/hscn-consumer-handbook/service-levels-and-incident-severity-classification)). Be pragmatic and don't overengineer against the requirements: targeting unnecessary levels of resilience is wasteful and reduces the team's ability to deliver functionality.
+    * Understand the implications of the designated service level for your service. NHS Digital service are classified as bronze, silver, gold or platinum and each has specific availability and recoverability requirements.
+    * Determine [service level indicators and objectives](https://landing.google.com/sre/sre-book/chapters/service-level-objectives/) (SLIs and SLOs) to represent how these requirements apply specifically to your service. Also consider what else is important to users, and add SLIs and SLOs for these other aspects of service quality, such as response times and error rates.
+    * Use SLIs and SLOs to drive monitoring and alerting requirements.
```
Comment (andyblundell) [inline practices/service-reliability.md:13]: ... and potentially to drive scaling as well, i.e. consider autoscaling based on the performance of the things the users care most about (in combination with other factors obviously, like CPU load)

File: practices/service-reliability.md [hunk 2]
```diff
@@ -8,8 +8,12 @@
...
-    * Agree an incident severity classification and the response which each level requires in terms of communication and resolution (example: [HSCN Consumer Handbook](https://digital.nhs.uk/services/health-and-social-care-network/hscn-consumer-handbook/service-levels-and-incident-severity-classification)). Be pragmatic and don't overengineer against the requirements: targeting unnecessary levels of resilience is wasteful and reduces the team's ability to deliver functionality.
+    * Understand the implications of the designated service level for your service. NHS Digital service are classified as bronze, silver, gold or platinum and each has specific availability and recoverability requirements.
+    * Determine [service level indicators and objectives](https://landing.google.com/sre/sre-book/chapters/service-level-objectives/) (SLIs and SLOs) to represent how these requirements apply specifically to your service. Also consider what else is important to users, and add SLIs and SLOs for these other aspects of service quality, such as response times and error rates.
```
Comment (andyblundell) [inline practices/service-reliability.md:12]: Can we emphasise that this is a collaborative exercise across the wider product team, not just an engineering activity?

File: practices/service-reliability.md [hunk 3]
```diff
@@ -8,8 +8,12 @@
...
+    * Use SLIs and SLOs to drive monitoring and alerting requirements.
+    * Consider what constitutes [meaningful availability](https://www.usenix.org/system/files/nsdi20spring_hauer_prepub.pdf) for your service, and the ways in which the service could fall back to degraded states to avoid total loss of availability, e.g. switching to read only mode, serving some proportion of demand only using [load shedding](https://aws.amazon.com/builders-library/using-load-shedding-to-avoid-overload/), serving some classes of demand but not others, disabling some features.
+    * Agree an incident severity classification and the response which each level requires in terms of communication and resolution (example: [HSCN Consumer Handbook](https://digital.nhs.uk/services/health-and-social-care-network/hscn-consumer-handbook/service-levels-and-incident-severity-classification)).
```
Comment (andyblundell) [inline practices/service-reliability.md:15]: Maybe combine this point with "Understand the implications of the designated service level for your service"? NHSD has a set of incident classifications, and the responses required for the different classifications are driven by the metallic scale.

File: practices/service-reliability.md (hunk 1 above)
Comment (ivorc) [reply practices/service-reliability.md:13]: Good call. We already have something very similar in the cloud services one, but it doesn't hurt to have it here too.

**Review by andyblundell — APPROVED** [review]

**Pull request merged**